torch/distributed/device_mesh.py

```diff
@@ -179,6 +180,109 @@ def create_flatten_mesh(
 
             return res_flattened_mesh
 
+        def create_unflatten_mesh(
```

Comment:
I must admit, this is a bit more complicated than I was hoping lol

Reply:
Do you still think the implementation is complicated?